estimagic/optimization/scipy_optimizers.py

```diff
@@ -0,0 +1,137 @@
+    and criterion function jointly, the user should provide a `criterion_and_derivative`
+    function that exploits the synergies in the calculation of criterion and gradient.
+
+    The lbfgsb algorithm is almost perfectly scale invariant. It is thus not necessary
+    to scale the parameters.
+
+    Below only details of the optional algorithm options are listed. For the mandatory
+    arguments see :ref:`internal_optimizer_interface`. For more background on those
+    options see :ref:`naming_conventions`.
+
+    Args:
+        relative_criterion_tolerance (float): Stop when the relative improvement between
+            two iterations is smaller than this. More formally, this is expressed as
+            ``f^k - f^{k+1})/max{|f^k|,|f^{k+1}|,1} <= relative_criterion_tolerance``
+        gradient_tolerance (float): Stop if all elements of the projected gradient are
+            smaller than this.
+        max_criterion_evaluations (int): If the maximum number of function evaluation is
+            reached, the optimization stops but we do not count this as convergence.
+        max_iterations (int): If the maximum number of iterations is reached, the
+            optimization stops, but we do not count this as convergence.
+        limited_memory_storage_length (int): Maximum number of saved gradients used to
+            approximate the hessian matrix.
+
+
+    Returns:
+        dict: See :ref:`internal_optimizer_output` for details.
+
+
+    """
+    algo_info = DEFAULT_ALGO_INFO.copy()
+    algo_info["name"] = "scipy_lbfgsb"
+    func = functools.partial(
+        criterion_and_derivative,
+        task="criterion_and_derivative",
+        algorithm_info=algo_info,
+    )
+
+    options = {
+        "maxcor": limited_memory_storage_length,
+        "ftol": relative_criterion_tolerance,
+        "gtol": gradient_tolerance,
+        "maxfun": max_criterion_evaluations,
+        "maxiter": max_iterations,
+        "maxls": max_line_search_steps,
+    }
+
+    res = scipy.optimize.minimize(
+        fun=func,
+        x0=x,
+        method="L-BFGS-B",
+        jac=True,
+        bounds=_get_scipy_bounds(lower_bounds, upper_bounds),
+        options=options,
+    )
+
+    return _process_scipy_result(res)
+
+
+def _process_scipy_result(scipy_results_obj):
+    # using get with defaults to access dict elements is just a safety measure
+    raw_res = {**scipy_results_obj}
+    processed = {
+        "solution_x": raw_res.get("x", None),
+        "solution_criterion": raw_res.get("fun", None),
+        "solution_derivative": raw_res.get("jac", None),
+        "solution_hessian": raw_res.get("hess", None),
+        "n_criterion_evaluations": raw_res.get("nfev", None),
+        "n_derivative_evaluations": raw_res.get("njac", None),
+        "n_iterations": raw_res.get("nit", None),
+        "success": raw_res.get("success", None),
+        "reached_convergence_criterion": None,
+        "message": raw_res.get("message", None),
```

Comment:
```suggestion
        "solution_x": raw_res.get("x"),
        "solution_criterion": raw_res.get("fun"),
        "solution_derivative": raw_res.get("jac"),
        "solution_hessian": raw_res.get("hess"),
        "n_criterion_evaluations": raw_res.get("nfev"),
        "n_derivative_evaluations": raw_res.get("njac"),
        "n_iterations": raw_res.get("nit"),
        "success": raw_res.get("success"),
        "reached_convergence_criterion": None,
        "message": raw_res.get("message"),
```

Reply:
And for all the mandatory fields which cannot be filled, you could implement `results = {**DEFAULT_NONE, **results}` after estimagic receives the results to make implementations easier.